includes/admin/class-wc-rest-payments-authorizations-controller.php

```diff
@@ -46,7 +46,7 @@ public function register_routes() {
 		);
 		register_rest_route(
 			$this->namespace,
-			'/' . $this->rest_base . '/(?P<payment_intent_id>\w+)',
+			'/' . $this->rest_base . '/(?P<payment_intent_id>[A-Za-z0-9_-]+)',
```

Comment:
It looks like we're not making any significant change here. In PHP `\w` means a "word" character, with the following explanation:

> A "word" character is any letter or digit or the underscore character, that is, any character which can be part of a Perl "word". The definition of letters and digits is controlled by PCRE's character tables, and may vary if locale-specific matching is taking place. For example, in the "fr" (French) locale, some character codes greater than 128 are used for accented letters, and these are matched by \w.

Therefore `\w` and `[A-Za-z0-9_-]` are mostly equivalent in ASCII, but differ in other encodings. However, the catch is that identifiers here should allow for UTF8 characters, whereas `[A-Za-z0-9_-]` limits them to the ASCII range.

Reply:
We are matching for `-` as well now with the updated identifier which `\w+` doesn't match with and are being more explicit. But yeah, I noticed that `\w+` doesn't match with `/` either. 
@marcinbot  WDYT about this?

Reply:
For now I have updated the API client instances where we are concatenating to build routes like in the original implementation and reverted the controller changes.